nimble/host/mesh/src/prov.c

```diff
@@ -237,7 +237,7 @@ static void reset_link(void)
 
 	/* Clear everything except the retransmit delayed work config */
 	memset(&link, 0, offsetof(struct prov_link, tx.retransmit));
-
+    link.conn_handle = BLE_HS_CONN_HANDLE_NONE;
```

Comment:
I think it should be guarded with #if (MYNEWT_VAL(BLE_MESH_PB_GATT))

also please check indentation here